pipeline/vars/common.groovy

```diff
@@ -148,33 +148,55 @@ def runTestSuite() {
     return executeTest(cmd, instanceName)
 }
 
-def sendEMail(def subjectPrefix,def test_results) {
-    /*
-        Send an email notification.
-    */
-    def body = readFile(file: "pipeline/vars/emailable-report.html")
-    body += "<body><u><h3>Test Summary</h3></u><br />"
-    body += "<p>Logs are available at ${env.BUILD_URL}</p><br />"
-    body += "<table><tr><th>Test Suite</th><th>Result</th>"
-
-    for (test in test_results) {
-        def res = "PASS"
-        if (test.value != 0) {
-            res = "FAIL"
+def fetchEmailBodyAndReceiver(def subjectPrefix, def test_results) {
+    def to_list = "ceph-qe-list@redhat.com"
+    def jobStatus = "stable"
+    def failureCount = 0
+    
+    def body = ""
+
+    if(subjectPrefix == "Tier-0"){
+        body += "<table><tr><th>Test Suite</th><th>Result</th>"
+        for (test in test_results) {
+            def res = "PASS"
+            if (test.value != 0) {
+                res = "FAIL"
+                failureCount += 1
+            } 
+
+            body += "<tr><td>${test.key}</td><td>${res}</td></tr>"
         }
+    }
+    else if(subjectPrefix == "Tier-1"){
+        body += "<table><tr><th>Jenkins Job Name</th><th>Result</th>"
+        for (test in test_results) {
+            if (test.value != "SUCCESS") {
+                failureCount += 1
+            }
 
-        body += "<tr><td>${test.key}</td><td>${res}</td></tr>"
+            body += "<tr><td>${test.key}</td><td>${test.value}</td></tr>"
+        }
     }
     body +="</table> </body> </html>"
 
-    def to_list = "ceph-qe-list@redhat.com"
-    def jobStatus = "stable"
-
-    if (1 in test_results.values()) {
+    if (failureCount > 0) {
         to_list = "cephci@redhat.com"
         jobStatus = 'unstable'
     }
+    return ["to_list" : to_list, "jobStatus" : jobStatus, "body" : body]
+}
 
+def sendEMail(def subjectPrefix,def test_results) {
+    /*
+        Send an email notification.
+    */
+    def body = readFile(file: "pipeline/vars/emailable-report.html")
+    body += "<body><u><h3>Test Summary</h3></u><br />"
+    body += "<p>Logs are available at ${env.BUILD_URL}</p><br />"
+    def params = fetchEmailBodyAndReceiver(subjectPrefix, test_results)
+    body += params["body"]
```

Comment:
You would need to close the `body` and `html` tags

Reply:
I have closed them in the method **fetchEmailBodyAndReceiver** itself. that should also be okay isn't it?